native/src/layout/flex.rs

```diff
@@ -124,19 +171,34 @@ where
                 max_main
             };
 
-            let (min_main, min_cross) =
-                axis.pack(min_main, axis.cross(limits.min()));
+            let (min_main, min_cross) = axis.pack(
+                min_main,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    axis.cross(limits.min())
+                },
+            );
 
-            let (max_main, max_cross) =
-                axis.pack(max_main, axis.cross(limits.max()));
+            let (max_main, max_cross) = axis.pack(
+                max_main,
+                if align_items == Align::Fill {
+                    cross
+                } else {
+                    max_cross
+                },
+            );
 
             let child_limits = Limits::new(
                 Size::new(min_main, min_cross),
                 Size::new(max_main, max_cross),
```

Comment:
While we are at it, we should probably rename these variables to be `*_width` and `*_height` like in the loop above.